### PR TITLE
Make pygments highlighter compatible with pygments 2

### PR DESCRIPTION
**Pull request by pankajp**
Changes same as done in ipython project
https://github.com/ipython/ipython/pull/6878
(IPython's highlighter is same as this)

### CODE REVIEW

**Comment by rkern** [comment]
+1 this is causing (ignorable but annoying) tracebacks in one of my applications.

File: pyface/ui/qt4/code_editor/pygments_highlighter.py [hunk 1]
```diff
@@ -21,6 +21,11 @@ def get_tokens_unprocessed(self, text, stack=('root',)):
     """ Split ``text`` into (tokentype, text) pairs.
 
         Monkeypatched to store the final stack on the object itself.
+
+        The `text` parameter this gets passed is only the current line, so to
```
Comment (itziakos) [inline pyface/ui/qt4/code_editor/pygments_highlighter.py]: `this` -> `that`

Comment (pankajp) [inline pyface/ui/qt4/code_editor/pygments_highlighter.py]: I'd prefer to leave it as is if that is fine, only to maintain exact match with the ipython code https://github.com/ipython/ipython/pull/6878/files .

Comment (rkern) [inline pyface/ui/qt4/code_editor/pygments_highlighter.py]: Fix both!

Comment (pankajp) [inline pyface/ui/qt4/code_editor/pygments_highlighter.py]: Changed this, that is out of scope of this project :)
Though i believe both usages are correct.
